tuist/tuist
# Add classPrefix support

### Short description 📝

This PR introduces changes that allows (optionally) defining "class prefix" for Xcode project. The "class prefix" is a prefix for class files Xcode generates when you create a project or class file. For example, it's used when Xcode generates source files from intent definition.

### How to test the changes locally 🧐

1. Set `classPrefix` argument in `Project.swift` manifest:
   ```swift
   import ProjectDescription

   let project = Project(
     name: "DemoApp",
     classPrefix: "TUIST",
     // ...
   )
   ```
2. Run `tuist generate` to generate the Xcode project.
3. Generated Xcode project should have the correct class prefix set, which can be verified in Xcode's inspector panel when the project is selected in project navigator:
   <img width="257" alt="Screenshot 2024-06-25 at 17 02 48" src="https://github.com/tuist/tuist/assets/1384684/19020c36-6d79-41b5-9355-b997b4631b45">

### Contributor checklist ✅


### Reviewer checklist ✅
